### ISSUE TITLE
Custom directory where images are stored

### ISSUE THREAD

**Issue by vsalvans** [DylanVann/react-native-fast-image#724]
My use case is that thousands of images must be already downloaded in order tu use them offline.

FastImage is awesome, in case the image is not downloaded and user is connected to Internet, it shows the image and caches it in a blink of an eye while scrolling. Great job! Thanks for this module!!

However I'm facing that the cache directory is flushed by iOS without warning (about 40000 image gone!) and I'd like to know if there is a possible option to use a different directory different than SDWebImage is using by default. I'd like to store or cache images in the documents folder rather than the cache directories.

I know is a special use case, just I wonder if someone knows if this module can or will have this feature or if anybody knows about a similar module that does that in a native way.

I did try a custom component that fits my requirements but its not native and when scrolling is make UX pretty bad.

Thanks!

**Comment by Pczek** [comment]
I'm currently researching this module for a similar use case. I want to have offline support of the images rendered by FastImage. Not sure if this is working already or not. By reading your post it seems that the caching directory could be flushed without notice, so the offline functionality might break unpredictbly. 🤔

**Comment by vsalvans** [comment]
What I'm doing is to have all images downloaded in the documents folder (using a sync custom service) where IOs can't delete unexpectedly . Then, the image component I'm using  is subscribed to connection changes(NetInfo) in order to pass a URL when online or the document file path when offline to FastImge. It's not ideal and the solution actually is not using the caching feature of FastImge, but it's really fast and low memory consumer which it's a big advantage when dealing with large scrolls of images. I'm hoping to have a custom cache folder configuration in FastImage ... or any other RN compoment.

**Comment by Pczek** [comment]
Yes, I also thought about building a custom solution as well, but, it's a custom solution. 🤷 
Feels more like a great addition to this library to me! :)

**Comment by haibert** [comment]
Guys do you know where the images are stored by default? I want to know because im not sure if its up to us delete the cache every once in while or does the library take care of this? If its up to us to take care of it Ill need to know where its stored.
